internal/controlplane/handlers_oauth.go

```diff
@@ -136,21 +138,32 @@ func (s *Server) GetAuthorizationURL(ctx context.Context,
 		return nil, err
 	}
 
+	var authorizationURL string
+	if slices.Contains(providerDef.AuthorizationFlows, db.AuthorizationFlowGithubAppFlow) {
+		gitHubAppConfig := s.cfg.Provider.GitHubApp
+		if gitHubAppConfig == nil {
+			return nil, status.Errorf(codes.Internal, "error getting GitHub App config: %s", err)
+		}
+		authorizationURL = fmt.Sprintf("https://github.com/apps/%v/installations/new?state=%v", gitHubAppConfig.AppName, state)
```

Comment:
Just a question here, is the GH app provider something we want to eventually support for e.g. enterprise github installations? I guess when we do this URL should come from the GH instance URL which would be set in the provider settings?

Reply:
I think so. I was spinning on this for a while, because at this section in the code the provider doesn't exist yet in the DB, so we would need to fetch the enterprise URL from somewhere else. I figured we could cross that bridge when we come to it and keep the flow simpler for now.

Reply:
Yeah - I think when we cross that bridge, we add the URL to the GHE server to the app provider-specific table, and write some more code.  😁